forge/src/multi_runner.rs

```diff
@@ -116,46 +103,57 @@ impl MultiContractRunnerBuilder {
         self.fuzzer = Some(fuzzer);
         self
     }
+
+    #[must_use]
+    pub fn evm_cfg(mut self, evm_cfg: Config) -> Self {
+        self.evm_cfg = Some(evm_cfg);
+        self
+    }
 }
 
 /// A multi contract runner receives a set of contracts deployed in an EVM instance and proceeds
 /// to run all test functions in these contracts.
-pub struct MultiContractRunner<E, S> {
-    /// Mapping of contract name to compiled bytecode, deployed address and logs emitted during
-    /// deployment
-    pub contracts: BTreeMap<String, (Abi, Address, Vec<String>)>,
+pub struct MultiContractRunner {
+    /// Mapping of contract name to Abi and creation bytecode
+    pub contracts: BTreeMap<String, (Abi, ethers::prelude::Bytes)>,
     /// Compiled contracts by name that have an Abi and runtime bytecode
     pub known_contracts: BTreeMap<String, (Abi, Vec<u8>)>,
     /// Identified contracts by test
     pub identified_contracts: BTreeMap<String, BTreeMap<Address, (String, Abi)>>,
     /// The EVM instance used in the test runner
-    pub evm: E,
+    pub evm_opts: EvmOpts,
+    /// The EVM revision config
+    pub evm_cfg: Config,
     /// The fuzzer which will be used to run parametric tests (w/ non-0 solidity args)
     fuzzer: Option<TestRunner>,
     /// The address which will be used as the `from` field in all EVM calls
     sender: Option<Address>,
-    /// Market type for the EVM state being used
-    state: PhantomData<S>,
 }
 
-impl<E, S> MultiContractRunner<E, S>
-where
-    E: Evm<S>,
-    S: Clone,
-{
+impl MultiContractRunner {
     pub fn test(
         &mut self,
-        filter: &impl TestFilter,
+        filter: &(impl TestFilter + Send + Sync),
     ) -> Result<BTreeMap<String, BTreeMap<String, TestResult>>> {
         // TODO: Convert to iterator, ideally parallel one?
         let contracts = std::mem::take(&mut self.contracts);
 
-        let init_state: S = self.evm.state().clone();
+        let vicinity = self.evm_opts.vicinity()?;
+        let backend = self.evm_opts.backend(&vicinity)?;
+
         let results = contracts
-            .iter()
+            .par_iter()
             .filter(|(name, _)| filter.matches_contract(name))
-            .map(|(name, (abi, address, logs))| {
-                let result = self.run_tests(name, abi, *address, logs, filter, &init_state)?;
+            .map(|(name, (abi, deploy_code))| {
+                // unavoidable duplication here?
+                let result = match backend {
+                    BackendKind::Simple(ref backend) => {
+                        self.run_tests(name, abi, backend, deploy_code.clone(), filter)?
+                    }
+                    BackendKind::Shared(ref backend) => {
+                        self.run_tests(name, abi, backend, deploy_code.clone(), filter)?
```

Comment:
cloning is a bit expensive, 
perhaps we can at a later stage either consume on `fn test` or remove the bytecode or wrap it in an arc

Reply:
`ethers::types::Bytes` is a wrapper around `bytes::Bytes`, which is cheap to clone https://docs.rs/bytes/latest/bytes/struct.Bytes.html